kratos/python/add_model_part_to_python.cpp

```diff
@@ -1016,8 +993,18 @@ void AddModelPartToPython(pybind11::module& m)
         .def("GetNodalSolutionStepTotalDataSize", &ModelPart::GetNodalSolutionStepTotalDataSize)
         .def("OverwriteSolutionStepData", &ModelPart::OverwriteSolutionStepData)
         .def("CreateNewNode", ModelPartCreateNewNode)
-        .def("CreateNewElement", ModelPartCreateNewElement)
-        .def("CreateNewCondition", ModelPartCreateNewCondition)
+        .def("CreateNewElement", [](ModelPart& rModelPart, const std::string ElementName, ModelPart::IndexType Id,
+            std::vector<ModelPart::IndexType> ElementNodeIds, ModelPart::PropertiesType::Pointer pProperties)
+            {return rModelPart.CreateNewElement(ElementName, Id, ElementNodeIds, pProperties); })
+        .def("CreateNewElement", [](ModelPart& rModelPart, const std::string ElementName, ModelPart::IndexType Id,
+            ModelPart::GeometryType::Pointer pGeometry, ModelPart::PropertiesType::Pointer pProperties)
+            {return rModelPart.CreateNewElement(ElementName, Id, pGeometry, pProperties);})
+        .def("CreateNewCondition", [](ModelPart& rModelPart, const std::string ConditionName, ModelPart::IndexType Id,
+            std::vector<ModelPart::IndexType> ConditionNodeIds, ModelPart::PropertiesType::Pointer pProperties)
+            {return rModelPart.CreateNewCondition(ConditionName, Id, ConditionNodeIds, pProperties); })
```

Comment:
@KratosMultiphysics/technical-committee  agrees that this is best implemented here, within the lambda, then in the model_part.. The point is that this is easily done without a special function from the c++ layer, although it is not easy to do from the python interface

also within the function we should have some DEBUG checks to verify that the nodes that belong to the geometry are also present in the modelpart.

Reply:
Hey @RiccardoRossi , I was checking the possibility of your suggested change. Now the thing is, it can be done within the lambda, but I don't see any advantage in this. If it is done here it can only be used from python. If we do it at the model part we can also use it in c++.

I have some use cases for using it from c++. I know that it is easier to avoid using it in c++, however, it is always connected with additional computation costs. I do not really see a reason why we need to add this additional efforts, just to reduce a function which rounds up the existing interfaces.

I personally do not really like to add logic in the python interface. This always creates confusion. But this is only my opinion, I know that this can be seen differently.